Filter/NodeSassFilter.php

```diff
@@ -142,4 +171,30 @@ public function filterLoad(AssetInterface $asset)
     public function filterDump(AssetInterface $asset)
     {
     }
+
+    protected function extractAndSaveSourceMap($output, $mapFileName)
+    {
+        // Extract mapping from node-sass output.
+        preg_match_all("/(.*)(\/\*\# sourceMappingURL=.*\*\/)(.*)/ms", $output, $matches);
+
+        if (count($matches) !== 4) {
+            return $output;
+        }
+
+        // Write source map to sourceMapLocation
+        $mappingFilePath = realpath($this->sourceMapLocation).DIRECTORY_SEPARATOR.$mapFileName;
+        file_put_contents($mappingFilePath, $matches[3][0]);
+
+        // Rewrite sourceMappingURL when sourceMapPublicDir is provided
+        $sourceMappingURL = ($this->sourceMapPublicDir)
+            ? $this->createSourceMappingUrlSegment($mapFileName)
+            : $matches[2][0];
+
+        return $matches[1][0].$sourceMappingURL;
+    }
+
+    protected function createSourceMappingUrlSegment($mapFileName)
```

Comment:
Why protected? Is this really necessary? It's used only in `extractAndSaveSourceMap`, maybe we can remove this?

Reply:
Yes, I'll inline the method.